packages/acceptance-tests/pkg-tests-specs/sources/commands/pack.test.js

```diff
@@ -234,6 +234,25 @@ describe(`Commands`, () => {
         expect(originalManifest.devDependencies[dependency]).toBe(`workspace:^1.0.0`);
         expect(originalManifest.peerDependencies[dependency]).toBe(`workspace:dependency`);
       }),
-    )
+    );
+
+    test(
+      `it should always include README (and its variants), even with a "files" field`,
+      makeTemporaryEnv({
+        files: [
+          `/lib/*.js`,
+        ],
+      }, async ({path, run, source}) => {
+        await fsUtils.writeFile(`${path}/lib/README`, `explaining lib`);
+        await fsUtils.writeFile(`${path}/README.md`, `explaining the package`);
+
+        await run(`install`);
+
+        const {stdout} = await run(`pack`, `--dry-run`);
+        await expect(stdout).not.toMatch(/lib\/README/);
```

Comment:
This does not match npm behavior currently. I [found a comment in `npm-packlist`](https://github.com/npm/npm-packlist/blob/270f534bc70dfb1d316682226332fd05e75e1b14/index.js#L159-L168) indicating that this was the original intent but now that people rely on it they don't want that breaking change. 

I wouldn't mind personally if this only includes top level files but since this also affects license files it might be safer to include nested files as well.

Reply:
Hm ... Given that it's a major release (and fairly consequent), I think it's reasonable to make a change like this (we should add a note in the changelog, though).

We can revisit if someone faces an issue because of it, but I'm not sure it's very likely.

(For the record, the rational for this top-level restriction is because of workspaces; if you pack a worktree for some reason, you might not want to also pack its children without any way to disable it)

Reply:
It's at least a breaking change that can be mitigated by explicitly including the files in the `files` entry.